### ISSUE TITLE
v7.1.2 doesn't parse `--file=message.txt` kind of `tag` argument correctly

### ISSUE THREAD

**Issue by ruohola** [EndBug/add-and-commit#179]
When using:

```
uses: EndBug/add-and-commit@v7
with:
  tag: 'v1.0.0 --annotate --file=message.txt'
```

the last argument gets parsed as `--file` `=message.txt`, when it should be parsed to just `--file=message.txt`.

This results in the following error:

```
> Tagging commit...
##[error]Error: fatal: could not open or read '=message.txt': No such file or directory

##[error]Error: fatal: could not open or read '=message.txt': No such file or directory

> Pushing commit to repo...
```

[**Demo of the buggy parsing**](https://www.typescriptlang.org/play?ts=4.2.3#code/PQKhCgAIUhZBDALgYwBYFMDOlEcgEwEsAzY9AJ3QDtFIAHJVTO9ZbeK-SecgcwFcAttUTYARgE9IlQQHsAboSq9IzeMiw5USbpUhVZtJZkL50kAI79DWKDADeAAQA2SgNaRUiRHUwAuYGBMRHU3BQpiZ1kAdwA6ZFlBYHhgADYAdgBGAAYAZlSAJkzgLNzcgBZU1IBfO0hHdAAPeEE6Z3Q6gANugCtMOsEkNABxQkQAQT5MAApOgFo54UxMeF5zACIAFVRCbF3uSAByKxt8Q8gllbX1yAXDDHIj7f3+TE1cDfuKG5PELHOFsRZLJIJkChVOgBKSAAXgAfJAANrrBaXVbodYAGkgWx2e3YR1+6DOFywVwx2JRcy+5CxOOe2Fe7zwAB11jS2ZZrH9MHSqUDZHywRV1gBdAZDVCjCZTabnSAKhWHaHwpGQcXQSDdTp1RyURD8chUbDjKi6cjwCTYkhaCjoQ7YAwXSWQMaHZzOSBiczoVqICR2YDgYj8KjIRCEWRmwYoKVjSa8GbBchKXh+VSIFPKaH2KCQBLG2iIWRuajYGGQADKmdTsQt0R1isg0wA-AAeWR0CNRhELFnRADUkLzCoAPs3zq225gLAjpiyWbFIOPEQA9Q6iyEwaHKkfL5s3Kf4WfNhdLlersVbyDQ9bDpvjqf1hEsytD8A6vMF4LScwV4ulsadboG06joNMwAspgA7ALw2KHLuX5Rj+lCYPwziIOmyapoioqwki4p5kCjzTN+tCCJAsjEBmWa8LEMZoOMHrTJQkI5nuqHoYgsR0K8qDTLEgkAPJiD0rDcfI8DOPwWDTIIsS8OQsj8L4y7jvY1SQrExCEBhFDTAAQsC7QcGxea1Hm+qGmanEYeAFnfrI7SxFEvByZK0oJjMhzyJksTZP5txzDp7QwmiayxIgjSIMqkIANxAA)

Caused by https://github.com/EndBug/add-and-commit/pull/174 which fixed another bug and created this regression.

**Comment by EndBug** [comment]
Hi, I've done some research, and this is what I've found.

The `process.argv` array is not generated by Node, but by the shell / OS, and so there's no way for me to recreate its exact behavior.

The same guy that made the core of the previous function made some adjustments, which I then turned in this function. It should parse pretty much every case and generate an arg array valid for simple-git.
[This](https://www.typescriptlang.org/play?ts=4.2.3#code/GYVwdgxgLglg9mAVAAgA4EMBOBnApgQUwHNsAKCAWwBMAZGMXALmWyk3qIEpkBvAKD7IhyCAlbJMuZAF5kAZTYcAdJnQB3AAaDhO0tp0HkAWiPJSAfgA8AGwREA8qigA+ZAB01Aak77Dw0m7YnsgAPjI+fsg+vjphepE6phaW2AAWcJhQji7uXhEJQoGeMcLRkXElhqQA5JUJydgAjq4BbkqhyADaAHrVALqcKPkF1cMJRXVlfhUFQqQARHWRyVTNZm5tHT3zA0NLwvNjkRORU4Yzs8mqaq49gfPVRn3e+4VBkwLlZla4mJgZrjccjOyA0KlwqGs6AguFIAHoinCiAAaZDVUYAbk+wmAGTM1lwUGQFGQcGAZkotHoUmC1TRnCUFHQUAgqXw1mspEknG4-EiBKJRBkyHsACMAFa4aBKXBgRS4MgUJREf4gVDYBnAGDWKC-UioGSuVCdACMAyxdQFXSgAE9ULhUQA3dDWPrCoidAAMfQtkWwahgLNSZlt9t5rxE6DwaN+-0w1UYEZ0UFS-zUyAY6YAon8MqQlfQqLgAB6cLGzSPR6pNBMRiBRqTVVa1is6G0wXDWKhdarO6wgXDVJ0u8GQ6GwhFuJGo9GcH1J4SiyToADW5dm9ar1xbraE7c73c6vZdA6HyD7893QiXuFX64Km8btjADicO93+67PbgTngYDPjxGNUngXgu17LiudYNmiaQZFkb6JleyCfoe1Q-rACAAUBIEupefgAL6+IRhF8M6mDILqrCECQwoKOwL4qOoWiJEYFAKtg6BEFI8wACqpDA2DIAJ7jzOgbjzGijQgHAupUHSbHYBxXGLIk2C+CYz6vhhYBonxwkgHggkptxMmpL8ElSTJCq1CxuJwMgJoAEwAMwACzqUYWoEtIClKbgShQMWUAeV5uCOdI8z4KgkIwPW2lonIaqoHB1QIvMvmcbg4kBUFixgpIY4wjUbj-jOowCKAkDxUyQYAOKBtRZCsPRRDMM1HDhjoohgOIkjYG1igvp0bqyMNvi4uR5BiESnQAPqotgbpkmgWB4I1pDtS+PISAqSioAZqQbWMkhQCAmA6X1fAkd12BwASSi2EQ+bMqy9VQOtlHvcQGqcEAA) is the interactive demo.

This is **fine**, but it has two main issues:
  - It's far from being perfect, and there will always be a case where a command works on your machine but doesn't with the action, because of the radical difference of the parsing
  - It's not the easiest thing to maintain: it relies on a Regex which I frankly don't understand very well, and the fact that I didn't code its core doesn't make it better. If it needs improvement it's not going to be easy

The ideal solution would be to rely on [`string-argv`](https://www.npmjs.com/package/string-argv), which seems to do the job well but doesn't correctly match `--message "This is \"a\" 'quoted' message"` ([demo](https://replit.com/@FedericoGrandi/NanoDeafeningDeadcode#index.js)).
I'll see if I can get to use this, maybe even dropping support for those type of strings.

I dunno, I'll think about it.

**Comment by ruohola** [comment]
>I dunno, I'll think about it.

No hurry at least in my end :) I can manage fine with either v7.1.1 or just by not using the entirely optional `=` between the argument name and the value.

**Comment by EndBug** [comment]
I've ended up deciding to rely on `string-argv`, just because it's way easier to maintain. I've also added a paragraph about argument parsing in the README, so that anybody having problems may find their answer.  
Thanks a lot for your contributions regarding this topic ❤️

**Comment by EndBug** [comment]
ℹ️ Fix published in `v7.2.0` (also `v7` and `latest`)
